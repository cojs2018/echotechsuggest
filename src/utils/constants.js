export const bookmarksAPI = 'https://5zcbdoz0h3.execute-api.eu-west-2.amazonaws.com/dev';