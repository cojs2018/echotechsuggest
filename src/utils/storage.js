import { writeFile } from 'fs';

export async function createBookmark(bookmarkUrl) {
    try {
        let bookmarkUrlList = require('../../bookmarkUrls.json');

        let bookmarkLocation = bookmarkUrl;
        if(!bookmarkLocation.includes('http')) {
            bookmarkLocation = `http://${bookmarkLocation}`;
        }

        const newUrlData = new URL(bookmarkLocation);

        const bookmarkJSON = {
            path: bookmarkUrl,
            url: newUrlData.toJSON(),
            createdAt: Date.now()
        }
        const articleName = `${newUrlData.pathname} - ${newUrlData.hostname}`;

        bookmarkUrlList[articleName] = bookmarkJSON;

        writeFile(
            '../../bookmarkUrls.json',
            JSON.stringify(bookmarkJSON, ' ', 4)
        )

        return {
            status: 200,
            message: 'Bookmark created successfully!'
        }
    }
    catch (error) {
        throw new Error(JSON.stringify({ 
            status: 500,
            message: 'Internal service error: Could not create bookmark!' 
        }));
    }
}

export async function listBookmarks() {
    const bookmarkUrlList = require('../../bookmarkUrls.json');
    return bookmarkUrlList;
}

export async function deleteBookmark(bookmarkName) {
    try {
        let bookmarkUrlList = require('../../bookmarkUrls.json');
        
        if(!bookmarkUrlList[bookmarkName]) {
            return {
                status: 200,
            }
        }

        delete bookmarkUrlList[bookmarkName]

        writeFile(
            '../../bookmarkUrls.json',
            JSON.stringify(bookmarkJSON, ' ', 4)
        )

        return {
            status: 200,
            message: 'Bookmark deleted successfully!'
        }
    }
    catch (error) {
        throw new Error(JSON.stringify({ 
            status: 500,
            message: 'Internal service error: Could not delete bookmark!' 
        }));
    }
}