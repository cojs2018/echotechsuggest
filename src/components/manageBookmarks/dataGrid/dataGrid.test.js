jest.mock('../../../utils/storage')

import { TestScheduler } from 'jest';
import React from 'react';
import render from 'react-test-renderer';
import { listBookmarks } from '../../../utils/storage';

import DataGrid from './dataGrid';

const getMockDate = () => {
    const randomTime = Math.floor(Math.random() * Date.now());
    const mockDate = new Date(randomTime);
    return mockDate.toUTCString();
};

const mockBookmarkUrls = [
    {
        bookmarkId: "0",
        articleName: "article0 - articlearchive",
        path: 'http://www.articlearchive.com/article0',
        url: new URL('http://www.articlearchive.com/article0'),
        createdAt: getMockDate(),
        tags: []
    },
    {
        bookmarkId: "1",
        articleName: "article1 - articlearchive",
        path: 'http://www.articlearchive.com/article1',
        url: new URL('http://www.articlearchive.com/article1'),
        createdAt: getMockDate(),
        tags: []
    },
    {
        bookmarkId: "2",
        articleName: "article2 - articlearchive",
        path: 'http://www.articlearchive.com/article2',
        url: new URL('http://www.articlearchive.com/article2'),
        createdAt: getMockDate(),
        tags: [ "tag 1" ]
    }
]

listBookmarks.mockReturnValue(new Promise((resolve, reject) => {
    resolve(mockBookmarkUrls);
}));

let rows = [];
let isLoading = false;
let isError = false;
let errorText = '';
let tablePage = 0;

const setRows = jest.fn(async newRows => rows = newRows);
const setIsLoading = jest.fn(async bool => isLoading = bool);
const setIsError = jest.fn(async bool => isError = bool);
const setErrorText = jest.fn(async newMessage => errorText = newMessage);
const setTablePage = jest.fn(async newTablePage => tablePage = newTablePage);

beforeEach(() => {
    React.useState = jest.fn()
        .mockReturnValueOnce([rows, setRows])
        .mockReturnValueOnce([isLoading, setIsLoading])
        .mockReturnValueOnce([isError, setIsError])
        .mockReturnValueOnce([errorText, setErrorText])
        .mockReturnValueOnce([tablePage, setTablePage]);

    React.useEffect = jest.fn()
        .mockImplementationOnce(fn => fn());
})

describe('DataGrid', () => {
    it('renders correctly', () => {
        const tree = render
            .create(<DataGrid />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
});