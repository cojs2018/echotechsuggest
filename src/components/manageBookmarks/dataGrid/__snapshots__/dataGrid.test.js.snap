// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 48,
            "paddingHorizontal": 16,
          },
          Object {
            "borderBottomColor": "rgba(0, 0, 0, 0.12)",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingVertical": 12,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.6)",
                },
              ],
            ]
          }
        >
          Article
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingVertical": 12,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.6)",
                },
              ],
            ]
          }
        >
          Created At
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingVertical": 12,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.6)",
                },
              ],
            ]
          }
        >
          Tags
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "paddingLeft": 16,
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "fontSize": 12,
                "marginRight": 44,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.6)",
              },
            ],
          ]
        }
      >
        1-10 of 0
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              Object {
                "opacity": 0.32,
              },
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û∞ÖÅ
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              Object {
                "opacity": 0.32,
              },
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û∞ÖÇ
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
